### PR TITLE
Shorten recv wait when there are buffered packets in banking stage

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
The banking stage takes long to find the next leader

#### Summary of Changes
Shortening the wait allows the banking stage to poll for next  leader more frequently

Fixes #

### CODE REVIEW

File: core/src/banking_stage.rs [hunk 1]
```diff
@@ -254,7 +254,18 @@ impl BankingStage {
                 .unwrap_or_else(|_| buffered_packets.clear());
             }
 
-            match Self::process_packets(&verified_receiver, &poh_recorder, recv_start) {
+            let recv_timeout = if !buffered_packets.is_empty() {
```
Comment (rob-solana) [inline core/src/banking_stage.rs]: wish the sense of this were inverted, is a little harder to read !is_empty()

Comment (pgarg66) [reply core/src/banking_stage.rs]: @rob-solana  would ```buffered_packets.len() > 0``` be better?

Comment (rob-solana) [reply core/src/banking_stage.rs]: anything but !is_empty() ;)

Comment (pgarg66) [reply core/src/banking_stage.rs]: @rob-solana It seems clippy prefers !is_empty.

https://buildkite.com/solana-labs/solana/builds/5235#996a7087-8231-40dc-b408-f8f24bd0a961
```

if buffered_packets.len() > 0 {
--
  | \|                ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: using `is_empty` is clearer and more explicit: `!buffered_packets.is_empty()`

```

Comment (rob-solana) [reply core/src/banking_stage.rs]: how about swapping the clauses?  if is_empty() default else special ?

**Pull request merged**